### PR TITLE
[PR1] DEV-45 Added Redux Store

### PR DESCRIPTION
**Pull request by smorsua**
This PR adds the redux store where all the global state is stored. For now, it just stores the PodData. You'll see some commented code in DataService. This is a bad practice but it will be shortly added in the next feature, so instead of removing it it's commented so you can ignore it.

### CODE REVIEW

File: src/models/PodData/Board.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export class Board {
+  name: string;
+  packets: Packet[];
+
+  constructor(name: string, packets: Packet[]) {
+    this.name = name;
+    this.packets = packets;
+  }
+
+  // public updatePacket(id: number) {
```
Comment (jmaralo) [inline src/models/PodData/Board.ts:12]: commented code

File: src/models/PodData/PodData.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  constructor(boards: Board[]) {
+    this.boards = boards;
+  }
+
+  public setLastBatchIDs(updates: PacketUpdate[]) {
+    this.lastBatchIDs = updates.map((update) => update.id);
+  }
+
+  public update(updates: PacketUpdate[]) {
```
Comment (jmaralo) [inline src/models/PodData/PodData.ts:17]: shouldn't this receive a map? the back sends a map as the packet update where the key is the packet id and the values are the packets

File: src/services/DataService.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      }${import.meta.env.VITE_POD_DATA_DESCRIPTION_URL}`
+    )
+    .then((response) => {
+      return response.data as PodData;
+    });
+}
+
+export const DataService = ({ children }: any) => {
+  const dispatch = useDispatch();
+  // const packetUpdateSocket = useRef(
```
Comment (jmaralo) [inline src/services/DataService.tsx:21]: commented code

File: src/services/OrderService.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  getOrderDescriptions(): Promise<OrderDescription[]>;
+  sendOrder(order: Order): void;
+}
+
+export const OrderContext = createContext<IOrderService>({} as IOrderService);
+
+export const OrderService = ({ children }: any) => {
+  const orderSocket = useRef(
+    new WebSocket(
+      `ws://${process.env.SERVER_IP}:${process.env.SERVER_PORT}/wsOrder`
```
Comment (jmaralo) [inline src/services/OrderService.tsx:16]: all the request to the backend should be prefixed with "backend" (e.g. "ws://localhost:1234/backend/wsOrder")

**Review by jmaralo — CHANGES_REQUESTED** [review]
Most of the things that I've seen are related to server communication. We have to talk where we should change those conflicts

File: src/models/PodData/PodData.ts (hunk 2 above)
Comment (smorsua) [reply src/models/PodData/PodData.ts:17]: We should check how go serializes maps

Comment (jmaralo) [reply src/models/PodData/PodData.ts:17]: as JSON objects

**Review by lipezaballa — APPROVED** [review]

**Review by jmaralo — APPROVED** [review]

**Pull request merged**